src/x11/x.c

```diff
@@ -942,8 +958,12 @@ static void x_shortcut_init(struct keyboard_shortcut *ks)
         g_free(str_begin);
 }
 
-int x_get_scale(void) {
-        return 1;
+float x_get_scale(void) {
+        const struct screen_info *scr_info = get_active_screen();
+        float scale = MAX(1, scr_info->dpi/96.);
+        LOG_D("X11 dpi: %i", scr_info->dpi);
+        LOG_D("X11 scale: %f", scale);
+        return scale;
```

Comment:
Could you add a scale setting that overrides this and wl_get_scale?

Reply:
Done.